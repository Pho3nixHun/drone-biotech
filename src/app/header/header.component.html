<nav class="flex justify-center p-4" id="hero-section">
    <div class="w-11/12 md:w-3/4 flex flex-col md:flex-row justify-between items-center">
        <a href="#"><img src="../../assets/phoenix.jpg" alt="logo" width="50" height="50" class="mb-4 md:mb-0"></a>
        <ul class="flex flex-col md:flex-row space-y-4 md:space-y-0 md:space-x-10 text-center md:text-left">
            <li><a href="#" class="hover:text-gray-400 duration-150">Home</a></li>
            <li><a href="#" class="hover:text-gray-400 duration-150" (click)="scrollToSection('product-section')">Products</a></li>
            <li><a href="#" class="hover:text-gray-400 duration-150" (click)="scrollToSection('partner-section')">Partners</a></li>
            <li><a href="#" class="hover:text-gray-400 duration-150" (click)="scrollToSection('contact-section')" >Contacts</a></li>
        </ul>
    </div>
</nav>
