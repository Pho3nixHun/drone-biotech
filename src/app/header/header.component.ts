import { Component, inject } from '@angular/core';
import { HeroComponent } from '../hero/hero.component';
import { ScrollService } from '../scroll.service';

@Component({
  selector: 'app-header',
  standalone: true,
  imports: [HeroComponent],
  templateUrl: './header.component.html',
})
export class HeaderComponent {
  private readonly scroll = inject(ScrollService);

  scrollToSection(sectionId: string) {
    this.scroll.scrollToSection(sectionId);
  }
}
