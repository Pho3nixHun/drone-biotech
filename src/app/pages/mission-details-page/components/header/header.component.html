<div class="flex md:items-center justify-between space-x-10">
    <div class="flex space-x-2 md:items-center flex-col md:flex-row">
        <ng-content select="[app-status, p]" />
    </div>
    <ng-content select="a" />
</div>
