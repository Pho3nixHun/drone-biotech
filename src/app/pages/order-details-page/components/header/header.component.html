<div class="flex flex-col space-y-4">
    <div class="flex flex-row items-start justify-between">
        <div><ng-content select="h2" /></div>
        <div class="flex flex-row space-x-3 items-center">
            <ng-content select="[app-status, button]" />
        </div>
    </div>
    <ng-content select="app-summary-list" />
</div>
