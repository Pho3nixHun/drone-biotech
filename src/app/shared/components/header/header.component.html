<section class="p-4 flex justify-center">
  <div
    class="w-full space-y-4 flex flex-col justify-between items-center md:w-3/4 md:flex-row md:space-y-0"
  >
    <ng-content select="[slot=logo]"></ng-content>
    <ng-content select="[slot=nav]"></ng-content>
  </div>
</section>
