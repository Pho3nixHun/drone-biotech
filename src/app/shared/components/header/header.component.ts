import { Component } from '@angular/core';
import { HeroComponent } from '@components/hero/hero.component';
import { ScrollDirective } from '@directives/scroll.directive';

@Component({
  selector: 'app-header',
  standalone: true,
  imports: [HeroComponent, ScrollDirective],
  templateUrl: './header.component.html',
})
export class HeaderComponent {}
