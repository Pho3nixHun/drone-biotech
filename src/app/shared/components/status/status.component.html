<p class="font-medium rounded-full w-max px-3 py-1 flex items-center">
    <ng-content />
</p>
