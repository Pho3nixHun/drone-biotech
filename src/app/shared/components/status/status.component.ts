import { Component, input } from '@angular/core';
import { StatusVM } from './status.model';
import { NgClass } from '@angular/common';

@Component({
    selector: 'app-status',
    imports: [NgClass],
    templateUrl: './status.component.html',
})
export class StatusComponent {
    public vm = input.required<StatusVM>();
}
