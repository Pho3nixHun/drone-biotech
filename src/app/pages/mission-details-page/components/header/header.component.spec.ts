import { ComponentFixture, TestBed } from '@angular/core/testing';
import { Component } from '@angular/core';
import { HeaderComponent } from './header.component';
import { StatusComponent } from '@components/status/status.component';

@Component({
    imports: [HeaderComponent, StatusComponent],
    template: `
        <app-header>
            <app-status></app-status>
            <p>Should be projected</p>
            <a href="">Should be projected</a>
            <div href="">Should not be projected</div>
        </app-header>
    `,
})
class TestHostComponent {}

describe('HeaderComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should project the content and render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
