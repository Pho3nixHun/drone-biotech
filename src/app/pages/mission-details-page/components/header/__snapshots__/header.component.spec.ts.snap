// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should project the content and render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-header>
    <div
      class="flex md:items-center justify-between space-x-10"
    >
      <div
        class="flex space-x-2 md:items-center flex-col md:flex-row"
      >
        <app-status>
          <p
            class="font-medium rounded-full w-max px-3 py-1 flex items-center"
          />
        </app-status>
        <p>
          Should be projected
        </p>
      </div>
      <a
        href=""
      >
        Should be projected
      </a>
    </div>
  </app-header>
</div>
`;
